File: ION/packages/Nfinium/XFINIUM.PDF 5.4.0 Mobile Professional Edition Commercial Version/Samples/Common/DocumentPageByPageSave.cs
using System;
using System.IO;
using Xfinium.Pdf;
using Xfinium.Pdf.Graphics;
using Xfinium.Pdf.Redaction;

namespace Xfinium.Pdf.Samples
{
    /// <summary>
    /// DocumentPageByPageSave sample.
    /// </summary>
    public class DocumentPageByPageSave
    {
        /// <summary>
        /// Main method for running the sample.
        /// </summary>
        public static SampleOutputInfo[] Run(Stream output)
        {
            PdfFixedDocument document = new PdfFixedDocument();

            // Init the save operation
            document.BeginSave(output);

            PdfRgbColor color = new PdfRgbColor();
            PdfBrush brush = new PdfBrush(color);
            Random rnd = new Random();

            for (int i = 0; i < 3; i++)
            {
                PdfPage page = document.Pages.Add();
                page.Width = 1000;
                page.Height = 1000;

                for (int y = 1; y <= page.Height; y++)
                {
                    for (int x = 0; x < page.Width; x++)
                    {
                        color.R = (byte)rnd.Next(256);
                        color.G = (byte)rnd.Next(256);
                        color.B = (byte)rnd.Next(256);

                        page.Graphics.DrawRectangle(brush, x, y - 1, 1, 1);
                    }

                    if ((y % 100) == 0)
                    {
                        // Compress the graphics that have been drawn so far and save them.
                        page.Graphics.Compress();
                        page.SaveGraphics();
                    }
                }

                // Close the page graphics and save the page.
                page.Graphics.CompressAndClose();
                page.Save();
            }

            // Finish the document.
            document.EndSave();

            return null;
        }
    }
}

File: ION/packages/Nfinium/XFINIUM.PDF 5.4.0 Mobile Professional Edition Commercial Version/Samples/Common/DocumentPageByPageSave.vb
Imports System.IO
Imports Xfinium.Pdf
Imports Xfinium.Pdf.Graphics
Imports Xfinium.Pdf.Redaction

Namespace Xfinium.Pdf.Samples
	''' <summary>
	''' DocumentPageByPageSave sample.
	''' </summary>
	Public Class DocumentPageByPageSave
		''' <summary>
		''' Main method for running the sample.
		''' </summary>
		Public Shared Function Run(output As Stream) As SampleOutputInfo()
			Dim document As New PdfFixedDocument()

			' Init the save operation
			document.BeginSave(output)

			Dim color As New PdfRgbColor()
			Dim brush As New PdfBrush(color)
			Dim rnd As New Random()

			For i As Integer = 0 To 2
				Dim page As PdfPage = document.Pages.Add()
				page.Width = 1000
				page.Height = 1000

				For y As Integer = 1 To page.Height
					For x As Integer = 0 To page.Width - 1
						color.R = CByte(rnd.[Next](256))
						color.G = CByte(rnd.[Next](256))
						color.B = CByte(rnd.[Next](256))

						page.Graphics.DrawRectangle(brush, x, y - 1, 1, 1)
					Next

					If (y Mod 100) = 0 Then
						' Compress the graphics that have been drawn so far and save them.
						page.Graphics.Compress()
						page.SaveGraphics()
					End If
				Next

				' Close the page graphics and save the page.
				page.Graphics.CompressAndClose()
				page.Save()
			Next

			' Finish the document.
			document.EndSave()

			Return Nothing
		End Function
	End Class
End Namespace
